### PR TITLE
Classify data category dropdown

### PR DESCRIPTION
**Pull request by allisonking**
Visual part of https://github.com/ethyca/fides/issues/1077 (not integrated yet)

### Code Changes

* [x] Refactor the existing `DataCategoryInput` to make it more reusable
* [x] Create `ClassifiedDataCategoryInput` which is two select boxes combined
* [x] Cypress component tests

### Steps to Confirm

* [ ] Since this doesn't exist in the app yet, the only way to see this is through running cypress component tests. There is one test called `playground` which will let you just play with the component

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* Documentation Updated:
  - [ ] documentation complete, or draft/outline provided (tag docs-team to complete/review on this branch)
  - [ ] documentation issue created (tag docs-team to complete issue separately)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [x] Update `CHANGELOG.md`

### Description Of Changes

I thought I'd put up this PR up before attempting to integrate it into the app to make sure the behavior looks right. Given that https://github.com/ethyca/fidesctl-plus/issues/99 isn't in yet, can't integrate fully, but could perhaps get a little farther.

https://user-images.githubusercontent.com/24641006/191825888-7649b3ea-dbd7-45bc-bad9-ee3bbee92dc2.mov

### CODE REVIEW

File: clients/ctl/admin-ui/src/features/dataset/ClassifiedDataCategoryDropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  checked,
+  ...props
+}: Props) => {
+  const menuButtonProps: ButtonProps = {
+    size: "sm",
+    colorScheme: "complimentary",
+    borderRadius: "6px 0px 0px 6px",
+  };
+
+  const options = mostLikelyCategories
```
Comment (ssangervasi) [inline clients/ctl/admin-ui/src/features/dataset/ClassifiedDataCategoryDropdown.tsx:32]: The design is ambiguous, but I noticed that by having this component only receive the most-likely categories, it's a bit unintuitive what happens on the right-half when using the checkbox tree:



"Identifiable" doesn't show up in the Select, but "Non-identifiable" does. Maybe we should show "N/A", "Custom", or "Manual" for those that don't have a percent?

File: clients/ctl/admin-ui/src/features/dataset/ClassifiedDataCategoryDropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+import { MultiValue, Select } from "chakra-react-select";
+
+import { DataCategory } from "~/types/api";
+
+import {
+  DataCategoryDropdown,
+  Props as DataCategoryDropdownProps,
+} from "./DataCategoryInput";
+
+// TODO: just making up a structure until we have something real from the API
```
Comment (ssangervasi) [inline clients/ctl/admin-ui/src/features/dataset/ClassifiedDataCategoryDropdown.tsx:11]: Nice. The category fides key will come from the [classification label](https://github.com/ethyca/fides/pull/1113/files#diff-d65a475197936464f11d722a344a559f2be72a46338a922105707f64d3f4bf40R31) and the score -> confidence. It should be easy enough in another PR to make a selector that combines these into this interface.

**Review by ssangervasi — APPROVED** [review]

File: clients/ctl/admin-ui/src/features/dataset/ClassifiedDataCategoryDropdown.tsx (hunk 1 above)
Comment (allisonking) [reply clients/ctl/admin-ui/src/features/dataset/ClassifiedDataCategoryDropdown.tsx:32]: hmm yeah, it's a strange component. I'm wondering if it'll help once this dropdown gets put in with the actual form which will list all the selected at the bottom. actually, lemme do that now to see if it "feels" any better

**Comment by allisonking** [comment]
@ssangervasi after your question about the design, I decided to just integrate the component into the page (with a TODO for what the `mostLikelyCategories` actually are—currently hard coded to the first 5 categories). That turned out to be a little more involved than I thought it would be due to circular dependencies, so I did some refactoring. Then, the cypress component test for `DataCategoryInput` didn't work anymore since I added a call to redux, so I also configured cypress component testing to have access to our store. 

So that's a few extra commits, and would love if you could take one more look!

File: clients/ctl/admin-ui/src/features/dataset/DataCategoryInput.tsx [hunk 3]
```diff
@@ -33,11 +36,23 @@ const DataCategoryInput = ({
...
+                checked={checked}
+                onChecked={onChecked}
+              />
+            ) : (
+              <DataCategoryDropdown
+                dataCategories={dataCategories}
+                checked={checked}
+                onChecked={onChecked}
+              />
+            )}
```
Comment (ssangervasi) [inline clients/ctl/admin-ui/src/features/dataset/DataCategoryInput.tsx:55]: The two separate stacks of categories is definitely funky. I was imagining that the new `select` would show all of the `checked` categories, whether or not they were from the classifier (but it would highlight the classifier's confidence). That would make the `TaxonomyEntityTag` below unnecessary.

Comment (allisonking) [reply clients/ctl/admin-ui/src/features/dataset/DataCategoryInput.tsx:55]: ahh I see what you mean. I'll try it out!

**Comment by allisonking** [comment]
Updated:


https://user-images.githubusercontent.com/24641006/192624393-0681988b-d9a1-4151-9616-8138336f88d4.mov

File: clients/ctl/admin-ui/src/features/dataset/ClassifiedDataCategoryDropdown.tsx [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+        return 1;
+      }
+      if (b.confidence === undefined) {
+        return -1;
+      }
+      return a.fides_key.localeCompare(b.fides_key);
+    })
+    .map((c) => {
+      const confidence =
+        c.confidence === undefined ? "N/A" : `${c.confidence}%`;
```
Comment (ssangervasi) [inline clients/ctl/admin-ui/src/features/dataset/ClassifiedDataCategoryDropdown.tsx:60]: Could throw a `confidence.toFixed()` on there just in case the number is overly precise.

File: clients/ctl/admin-ui/src/features/dataset/ClassifiedDataCategoryDropdown.tsx [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+      const addedKey = newKeys.filter(
+        (newKey) => oldKeys.indexOf(newKey) === -1
+      )[0];
+      newChecked = [...checked, addedKey];
+    } else {
+      const removedKey = oldKeys.filter(
+        (oldKey) => newKeys.indexOf(oldKey) === -1
+      )[0];
+      newChecked = checked.filter((c) => c !== removedKey);
+    }
```
Comment (ssangervasi) [inline clients/ctl/admin-ui/src/features/dataset/ClassifiedDataCategoryDropdown.tsx:86]: There are enough indexOf's that my instinct is to use some Sets or Maps. If sort order is a concern, fortunately those classes preserve their order when converting to/from arrays.

File: clients/ctl/admin-ui/cypress/components/ClassifiedDataCategoryInput.cy.tsx [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+}));
+const MOST_LIKELY_CATEGORIES = CATEGORIES_WITH_CONFIDENCE.slice(0, 5);
+
+describe("ClassifiedDataCategoryDropdown", () => {
+  it("can render checked categories", () => {
+    const onCheckedSpy = cy.spy().as("onCheckedSpy");
+    const selectedClassifiedCategory = MOST_LIKELY_CATEGORIES[0];
+    cy.mount(
+      <ClassifiedDataCategoryDropdown
+        dataCategories={MOCK_DATA_CATEGORIES as DataCategory[]}
```
Comment (ssangervasi) [inline clients/ctl/admin-ui/cypress/components/ClassifiedDataCategoryInput.cy.tsx:19]: If the definition of `MOCK_DATA_CATEGORIES` had the api type, or used `as const`, these casts wouldn't be necessary.

**Review by ssangervasi — APPROVED** [review]
👍

File: clients/ctl/admin-ui/cypress/components/ClassifiedDataCategoryInput.cy.tsx (hunk 6 above)
Comment (allisonking) [reply clients/ctl/admin-ui/cypress/components/ClassifiedDataCategoryInput.cy.tsx:19]: oh right! I meant to talk about this with you. So the openapi type has type `parent_key?: string`. however, it turns out `parent_key = null` in real life! I'm not sure if this is just the typescript generator not being good about null vs undefined or something about our backend definition, but that's why I had this cast

Comment (ssangervasi) [reply clients/ctl/admin-ui/cypress/components/ClassifiedDataCategoryInput.cy.tsx:19]: Still digging into the OpenApi schema, but the answer may be that we should use [response_model_exclude_none](https://fastapi.tiangolo.com/tutorial/response-model/?h=response_model_exclude_none#use-the-response_model_exclude_unset-parameter) on our routes to exclude optionals instead of passing nulls. Applying that to all crud endpoints would be very nice for consistency.

As-is, I think the TS generator is _technically_ correct because if you look directly at the OpenAPI schema it says the value is optional (not required) and does _not_ say the type can be `null`:
Screen Shot 2022-09-30 at 16 59 19
(`*` by a field indicates it is required)

I don't know of a way to get fastapi/pydantic to say the type can be `null` -- [this](https://fastapi.tiangolo.com/it/tutorial/query-params-str-validations/#required-with-none) didn't do it when I tried -- and I would rather exclude them anyway.
